### PR TITLE
SRIOV provider fine tunes: Wait for pods to be ready during sriov-operator deployment

### PR DESCRIPTION
**Pull request by ormergi**
sriov-network-operator deployment includes creating the pods:
- sriov-network-operator
- sriov-network-config-daemon
- operator-webhook
- network-resources-injector

At the current state we deploy 'sriov-network-operator' and create certificates & Secrets on the same time,
where the sriov-operator deployment runs in the background.
The problem is that it put more pressure on etcd and affect its performance (**might** be a cause for etcd for timeouts)

This PR make the deployment sequential by adding waits between the deployment 
of the operator and the certificates & Secrets creation, hopefully it will reduce the pressure 
on etcd pods.

### CODE REVIEW

**Comment by oshoval** [comment]
I think that since i was evolved in the research
https://github.com/kubevirt/kubevirtci/pull/473#discussion_r519932789
the minimal is to cc me as well

**Review by oshoval — CHANGES_REQUESTED** [review]
Please see the problem fixed at
https://github.com/kubevirt/kubevirtci/pull/481

Its not related to this PR, but to the usage of "wait pods" in general, which
existed before this PR.
Since this PR adds additional usage of "wait pods", it will need to be adapted as well.

EDIT
One possible option is that you first merge this PR, as is, once you get approval.
And once #481 is merged you can adapt the new function if you want, unless you are ok with waiting please for #481.
As long as it will be before we bump to kubevirt, all good imo, since we know to track it.

**Comment by ormergi** [comment]
> Please see the problem fixed at
> #481
> 
> Its not related to this PR, but to the usage of "wait pods" in general, which
> existed before this PR.
> Since this PR adds additional usage of "wait pods", it will need to be adapted as well.
> 
> EDIT
> One possible option is that you first merge this PR, as is, once you get approval.
> And once #481 is merged you can adapt the new function if you want, unless you are ok with waiting / rebasing please #481.
> As long as it will be before we bump to kubevirt, all good imo, since we know to track it.

The first PR that will be merge then the other will adapt, fair enough?

**Comment by oshoval** [comment]
> > Please see the problem fixed at
> > #481
> > Its not related to this PR, but to the usage of "wait pods" in general, which
> > existed before this PR.
> > Since this PR adds additional usage of "wait pods", it will need to be adapted as well.
> > EDIT
> > One possible option is that you first merge this PR, as is, once you get approval.
> > And once #481 is merged you can adapt the new function if you want, unless you are ok with waiting / rebasing please #481.
> > As long as it will be before we bump to kubevirt, all good imo, since we know to track it.
> 
> The first PR that will be merge then the other will adapt, fair enough?

As long as the logic of #481 isn't copied to this PR (especially fixing the pod restart problem), yes.
Fair enough?

**Comment by oshoval** [comment]
> > Please see the problem fixed at
> > #481
> > Its not related to this PR, but to the usage of "wait pods" in general, which
> > existed before this PR.
> > Since this PR adds additional usage of "wait pods", it will need to be adapted as well.
> > EDIT
> > One possible option is that you first merge this PR, as is, once you get approval.
> > And once #481 is merged you can adapt the new function if you want, unless you are ok with waiting / rebasing please #481.
> > As long as it will be before we bump to kubevirt, all good imo, since we know to track it.
> 
> The first PR that will be merge then the other will adapt, fair enough?

Fair enough.

**Comment by ormergi** [comment]
/close 

No longer needed https://github.com/kubevirt/kubevirtci/pull/492

**Pull request closed without merging**
